app/Services/Manager.php

```diff
@@ -80,4 +97,62 @@ public function collectContestInfo($data)
 
         return $contest;
     }
+
+    /**
+     * Get user signup for a specific campaign and run. If only user id is
+     * provided, the send back all user signups.
+     *
+     * @param  string $id
+     * @param  string $campaign    Campaign ID
+     * @param  string $campaignRun Campaign Run ID
+     *
+     * @return object $signups
+     */
+    public function getUserSignup($id, $campaign = null, $campaignRun = null)
+    {
+        $signups = $this->northstar->getUserSignups($id, $campaign);
+
+        // Only return the sign up record for the run that was specified.
+        if ($campaignRun) {
+            foreach ($signups as $key => $signup) {
+                if ($signup->campaign_run->id == $campaignRun) {
+                    $signups = $signups[$key];
+                }
+            }
+        }
+
+        return $signups;
+    }
+
+    /**
+     * Get a user's signup/reportback activity for a
+     * competition or waiting room.
+     *
+     * @param  string $id  User ID
+     * @param  \Gladiator\Models\Competition|WaitingRoom $model
+     *
+     * @return object $reportback
+     */
+    public function getUserActivity($id, $model)
+    {
+        $campaign = $model->contest->campaign_id;
+        $campaign_run = $model->contest->campaign_run_id;
+
+        $signup = $this->getUserSignup($id, $campaign, $campaign_run);
+
+        if ($signup && $signup->reportback) {
+            // Provide the admin URL to the reportback.
+            $signup->reportback->admin_url = env('PHOENIX_URI') . '/admin/reportback/' . $signup->reportback->id;
+
+            // Format the update timestamp
+            $signup->reportback->updated_at = new Carbon($signup->reportback->updated_at);
+            $signup->reportback->updated_at = $signup->reportback->updated_at->format('Y-m-d');
```

Comment:
Could this be combined w/ the line above?

```
 $signup->reportback->updated_at = new Carbon($signup->reportback->updated_at)->format('Y-m-d');
```

Not actually sure if that works...